apis/apps/v1beta1/statefulset_types.go

```diff
@@ -299,6 +305,37 @@ type StatefulSetSpec struct {
 	Ordinals *StatefulSetOrdinals `json:"ordinals,omitempty"`
 }
 
+type ReserveOrdinal []intstr.IntOrString
+
+func (r ReserveOrdinal) GetIntSet() sets.Set[int] {
```

Comment:
consider moving the function to the utility package, the api package should not contain func other than simple deepcopys